portfolio/src/main/java/com/google/sps/data/BlogMessage.java

```diff
@@ -34,7 +34,11 @@ public BlogMessage(long id, String tag, String message,
     this.message = message;
     this.nickname = nickname;
     this.email = email;
-    this.messageReplies = messageReplies;
+    if (messageReplies != null) {
+      this.messageReplies = messageReplies;   
+    } else {
+       this.messageReplies = new ArrayList<BlogMessage>();
```

Comment:
I'm not sure if it's conventional to check one member for null in a constructor with many members or if it's clear to someone that doesn't know why you're doing this (Marti/Peizhao thoughts?) 
But to avoid this, and since messageReplies contain BlogMessages, we can define a default constructor with no params and use it for messageReplies when we're calling the constructor with params.

Reply:
I'm not very familiar with java, but this should be OK. Just note in the comments for the constructor which parameters can be null.

Reply:
ok